packages/cardpay-subgraph/src/mappings/register-reward-program-handler.ts

```diff
@@ -0,0 +1,33 @@
+import { RewardProgramRegistrationFee } from '../../generated/RewardProgram/RegisterRewardProgramHandler';
+import { log } from '@graphprotocol/graph-ts';
+
+import { RewardProgramRegistrationPayment } from '../../generated/schema';
+import { toChecksumAddress, makePrepaidCardPayment, makeToken, makeTransaction } from '../utils';
+
+export function handleRewardProgramRegistrationFee(event: RewardProgramRegistrationFee): void {
+  makeTransaction(event);
+  let prepaidCard = toChecksumAddress(event.params.prepaidCard);
+  let admin = toChecksumAddress(event.params.admin);
+  let rewardProgramID = toChecksumAddress(event.params.rewardProgramID);
+  let issuingToken = makeToken(event.params.issuingToken);
+  let txnHash = event.transaction.hash.toHex();
+  log.info('====prepaidCard {}', [prepaidCard]);
+  log.info('====admin {}', [admin]);
+  log.info('====rewardProgramID {}', [rewardProgramID]);
+
+  makePrepaidCardPayment(
+    event,
+    prepaidCard,
+    null,
+    issuingToken,
+    event.params.issuingTokenAmount,
+    event.params.spendAmount
+  );
+  let entity = new RewardProgramRegistrationPayment(rewardProgramID);
```

Comment:
this is a really bad ID. you can have multiple payments per reward program right? in this case you'll stomp on all those payments. Consider using txnHash for entities that are types of events--in my mind a "payment" is a type of event. (although make sure to think about entities that can appear multiple times in a single txn--txnHashes alone won't work for those).

Reply:
Hmmm sure. I think this is a good suggestion since txn is the more unique id. I will implement the change.